private void close(Closeable in) {
    if (in != null) {
        try {
            in.close();
        } catch (IOException e) {
            getExceptionListener().exceptionThrown(e);
        }
    }
}